libbeat/outputs/elasticsearch/client.go

```diff
@@ -76,15 +81,38 @@ type clientSettings struct {
 type bulkResultStats struct {
 	acked        int // number of events ACKed by Elasticsearch
 	duplicates   int // number of events failed with `create` due to ID already being indexed
-	fails        int // number of failed events (can be retried)
-	nonIndexable int // number of failed events (not indexable)
+	fails        int // number of events with retryable failures.
+	nonIndexable int // number of events with permanent failures.
+	deadLetter   int // number of failed events ingested to the dead letter index.
 	tooMany      int // number of events receiving HTTP 429 Too Many Requests
 }
 
+type bulkResult struct {
+	// A connection-level error if the request couldn't be sent or the response
+	// couldn't be read. This error is returned from (*Client).Publish to signal
+	// to the pipeline that this output worker needs to be reconnected before the
+	// next Publish call.
+	connErr error
+
+	// The array of events sent via bulk request. This excludes any events that
+	// had encoding errors while assembling the request.
+	events []publisher.Event
+
+	// The http status returned by the bulk request.
+	status int
+
+	// The API response from Elasticsearch.
+	response eslegclient.BulkResponse
+}
+
 const (
 	defaultEventType = "doc"
 )
 
+var bulkRequestParams = map[string]string{
```

Comment:
Can we get some comments explaining this ES param? It's a bit odd without any context

Reply:
Done